test-integration/filter-gro.js

```diff
@@ -0,0 +1,173 @@
+/* global artifacts */
+
+const { BN } = require("bn.js");
+const { assert } = require("chai");
+const { deployArgent } = require("../utils/argent-deployer.js");
+const utils = require("../utils/utilities.js");
+
+const DepositHandler = artifacts.require("GroDepositHandler");
+const WithdrawHandler = artifacts.require("GroWithdrawHandler");
+const Controller = artifacts.require("GroController");
+const DepositFilter = artifacts.require("GroDepositFilter");
+const WithdrawFilter = artifacts.require("GroWithdrawFilter");
+const ERC20 = artifacts.require("TestERC20");
+
+const ethAmount = web3.utils.toWei("0.01");
+const daiAmount = web3.utils.toWei("10");
+
+contract("Gro Filter", (accounts) => {
+  let argent;
+  let wallet;
+
+  let gvt;
+  let pwrd;
+  let depositHandler;
+  let withdrawHandler;
+
+  before(async () => {
+    argent = await deployArgent(accounts);
+
+    gvt = await ERC20.at("0x3ADb04E127b9C0a5D36094125669d4603AC52a0c");
+    pwrd = await ERC20.at("0xf0a93d4994b3d98fb5e3a2f90dbc2d69073cb86b");
+
+    const controller = await Controller.at("0xCC5c60A319D33810b9EaB9764717EeF84deFB8F4");
+    await controller.switchEoaOnly(false, { from: "0xdc954086cf07f3889f186118395bad186179ac77" });
+
+    depositHandler = await DepositHandler.at("0xB7207Ea9446DcA1dEC1c1FC93c6Fcdf8B4a44F40");
+    withdrawHandler = await WithdrawHandler.at("0x641bEFA4dB601578A64F0Fc1f4E89E9869268Fe7");
+
+    const depositFilter = await DepositFilter.new();
+    const withdrawFilter = await WithdrawFilter.new();
+
+    await argent.dappRegistry.addDapp(0, depositHandler.address, depositFilter.address);
+    await argent.dappRegistry.addDapp(0, withdrawHandler.address, withdrawFilter.address);
+  });
+
+  beforeEach(async () => {
+    wallet = await argent.createFundedWallet({
+      DAI: daiAmount,
+      WETH: ethAmount,
+    });
+  });
+
+  const deposit = async (isPwrd) => argent.multiCall(wallet, [
```

Comment:
We could wrap this in your `checkBalances` function. Idem for the withdraw functions.